state/state.go

```diff
@@ -114,26 +119,95 @@ func detectPartitionByFindmnt(b *block.Partition) PartitionState {
 	}
 }
 
-func detectBoot() Boot {
+func detectBoot(logger zerolog.Logger) Boot {
+	logger.Info().Msg("detecting boot state")
 	cmdline, err := os.ReadFile("/proc/cmdline")
 	if err != nil {
+		logger.Debug().Err(err).Msg("Error reading /proc/cmdline file " + err.Error())
 		return Unknown
 	}
+
 	cmdlineS := string(cmdline)
+
+	if DetectUKIboot(cmdlineS) {
+		logger.Debug().Msg("Detected uki boot")
+		return getUKIBootState(logger)
+	}
+
+	return getNonUKIBootState(cmdlineS)
+}
+
+func getUKIBootState(logger zerolog.Logger) Boot {
+	if !EfiBootFromInstall(logger) {
+		return LiveCD
+	}
+
+	currentEntryBytes, err := os.ReadFile(UEFICurrentEntryFile)
+	if err != nil {
+		logger.Debug().Err(err).Msg(fmt.Sprintf("Error reading %s file %s", UEFICurrentEntryFile, err.Error()))
+		return Unknown
+	}
+
+	// Create a regular expression to remove non-printable characters
+	regex := regexp.MustCompile("[[:cntrl:]]")
+	currentEntry := regex.ReplaceAllString(string(currentEntryBytes), "")
+
+	logger.Debug().Msg("Current entry: " + currentEntry)
+
 	switch {
-	case strings.Contains(cmdlineS, "COS_ACTIVE"):
+	case strings.HasSuffix(currentEntry, "active.conf"):
 		return Active
-	case strings.Contains(cmdlineS, "COS_PASSIVE"):
+	case strings.HasSuffix(currentEntry, "passive.conf"):
 		return Passive
-	case strings.Contains(cmdlineS, "COS_RECOVERY"), strings.Contains(cmdlineS, "COS_SYSTEM"), strings.Contains(cmdlineS, "recovery-mode"):
+	case strings.HasSuffix(currentEntry, "recovery.conf"):
 		return Recovery
-	case strings.Contains(cmdlineS, "live:LABEL"), strings.Contains(cmdlineS, "live:CDLABEL"), strings.Contains(cmdlineS, "netboot"):
+	default:
+		return Unknown
+	}
+}
+
+func getNonUKIBootState(cmdline string) Boot {
+	switch {
+	case strings.Contains(cmdline, "COS_ACTIVE"):
+		return Active
+	case strings.Contains(cmdline, "COS_PASSIVE"):
+		return Passive
+	case strings.Contains(cmdline, "COS_RECOVERY"), strings.Contains(cmdline, "COS_SYSTEM"), strings.Contains(cmdline, "recovery-mode"):
+		return Recovery
+	case strings.Contains(cmdline, "live:LABEL"), strings.Contains(cmdline, "live:CDLABEL"), strings.Contains(cmdline, "netboot"):
 		return LiveCD
 	default:
 		return Unknown
 	}
 }
 
+// Detects if we are on uki mode
+func DetectUKIboot(cmdline string) bool {
+	Log.Info().Msg("checking cmdline for uki:" + cmdline)
+	return strings.Contains(cmdline, "rd.immucore.uki")
+}
```

Comment:
This is very similar to this: https://github.com/kairos-io/kairos-agent/blob/5f6bb7884f5c2cae6c9b7e54895be7616ba0735b/pkg/utils/common.go#L505-L511

we should consider merging the 2, or calling one from the other.